AreaShop/src/main/java/me/wiefferink/areashop/ASLanguageManager.java

```diff
@@ -0,0 +1,164 @@
+            message = currentLanguage.get(key);
+        } else {
+            message = defaultLanguage.get(key);
+        }
+        if(message == null) {
+            Log.warn("Did not find message '" + key + "' in the current or default language");
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(message);
+    }
+
+    private void saveDefaults() {
+        // Create the language folder if it not exists
+        File langFolder;
+        if(!languageFolder.exists()) {
+            if(!languageFolder.mkdirs()) {
+                Log.warn("Could not create language directory: " + languageFolder.getAbsolutePath());
+                return;
+            }
+        }
+
+        try {
+            // Read jar as ZIP file
+            File jarPath = new File(plugin.getClass().getProtectionDomain().getCodeSource().getLocation().toURI());
+            ZipFile jar = new ZipFile(jarPath);
+            Enumeration<? extends ZipEntry> entries = jar.entries();
+
+            // Each entry is a file or directory
+            while(entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+
+                // Filter to YAML files in the language directory
+                if(!entry.isDirectory() && entry.getName().startsWith(jarLanguagePath+"/") && entry.getName().endsWith(".yml")) {
+                    // Save the file to disk
+                    File targetFile = new File(languageFolder.getAbsolutePath() + File.separator + entry.getName().substring(entry.getName().lastIndexOf("/")));
+                    try(
+                            InputStream input = jar.getInputStream(entry);
+                            OutputStream output = new FileOutputStream(targetFile)
+                    ) {
+                        int read;
+                        byte[] bytes = new byte[1024];
+                        while((read = input.read(bytes)) != -1) {
+                            output.write(bytes, 0, read);
+                        }
+                    } catch(IOException e) {
+                        Log.warn("Something went wrong saving a default language file: " + targetFile.getAbsolutePath());
+                    }
+                }
+            }
+        } catch(URISyntaxException e) {
+            Log.error("Failed to find location of jar file:", ExceptionUtils.getStackTrace(e));
+        } catch(IOException e) {
+            Log.error("Failed to read zip file:", ExceptionUtils.getStackTrace(e));
+        }
+    }
+
+    private Map<String, List<String>> loadLanguage(String key) {
+        return loadLanguage(key, true);
+    }
+
+    /**
+     * Loads the specified language
+     * @param key	 The language to load
+     * @param convert try conversion or not (infinite recursion prevention)
+     * @return Map with the messages loaded from the file
+     */
+    private Map<String, List<String>> loadLanguage(String key, boolean convert) {
```

Comment:
I assume you've copied this straight from the LanguageManage super method, have you changed anything here? If you haven't you might as well refactor it to use the NIO since we're on Java 17 now.